src/core/ext/filters/client_channel/http_proxy_mapper.cc

```diff
@@ -229,6 +247,22 @@ absl::optional<std::string> HttpProxyMapper::MapName(
   return name_to_resolve;
 }
 
+absl::optional<grpc_resolved_address> HttpProxyMapper::MapAddress(
+    const grpc_resolved_address& address, ChannelArgs* args) {
+  auto proxy_address = GetAddressProxyServer(args);
+  if (!proxy_address.has_value()) {
+    return absl::nullopt;
+  }
+  auto address_string = grpc_sockaddr_to_string(&address, true);
+  if (!address_string.ok()) {
+    gpr_log(GPR_ERROR, "Unable to convert address to string: %s",
```

Comment:
This case should return nullopt.  If we can't convert the address to string form, we can't use the proxy, we shouldn't return the proxy address.  The way this code is written right now, it will tell gRPC to connect to the proxy address, but it won't actually configure the HTTP CONNECT handshaker.

Reply:
Let's not worry about testing it for now.  This is unlikely to be used in cases where the address isn't a known type anyway, since that basically happens only in the binder transport.  And this problem will eventually go away as part of #34172.